src/wkt/tests/null_value.rs

```diff
@@ -0,0 +1,149 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Test serialization and deserialization of `NullValue`.
+//!
+//! `NullValue` is an well-known type that represents a null JSON value. We need
+//! to verify it works well as part of a larger message or in enums.
+
+#[cfg(test)]
+mod test {
+    use google_cloud_wkt as wkt;
+    use serde_json::json;
+    type TestResult = std::result::Result<(), Box<dyn std::error::Error>>;
+
+    #[test]
+    fn test_serialize() -> TestResult {
+        let input = Value::new()
+            .set_test_name("test_serialize")
+            .set_null_value(wkt::NullValue);
+        let got = serde_json::to_value(&input)?;
+        let want = json!({
+            "testName": "test_serialize",
+            "nullValue": null
+        });
+        assert_eq!(got, want);
+
+        let input = Value::new()
+            .set_test_name("test_serialize")
+            .set_duration(wkt::Duration::clamp(123, 456));
+        let got = serde_json::to_value(&input)?;
+        let want = json!({
+            "testName": "test_serialize",
+            "duration": "123.000000456s",
+        });
+        assert_eq!(got, want);
+
+        let input = Value::new()
+            .set_test_name("test_serialize")
+            .set_number(123.5);
+        let got = serde_json::to_value(&input)?;
+        let want = json!({
+            "testName": "test_serialize",
+            "number": 123.5,
+        });
+        assert_eq!(got, want);
+        Ok(())
+    }
+
+    #[test]
+    fn test_deserialize() -> TestResult {
+        let input = json!({
+            "testName": "test_serialize",
+            "nullValue": null
+        });
+        let got = serde_json::from_value::<Value>(input)?;
+        let want = Value::new()
+            .set_test_name("test_serialize")
+            .set_null_value(wkt::NullValue);
+        assert_eq!(got, want);
+
+        let input = json!({
+            "testName": "test_serialize",
+            "duration": "123.000000456s",
+        });
+        let got = serde_json::from_value::<Value>(input)?;
+        let want = Value::new()
+            .set_test_name("test_serialize")
+            .set_duration(wkt::Duration::clamp(123, 456));
+        assert_eq!(got, want);
+
+        let input = json!({
+            "testName": "test_serialize",
+            "number": 123.5,
+        });
+        let got = serde_json::from_value::<Value>(input)?;
+        let want = Value::new()
+            .set_test_name("test_serialize")
+            .set_number(123.5);
+        assert_eq!(got, want);
+        Ok(())
+    }
+
+    // A test message, inspired by `google.firestore.v1.ValueType`.
```

Comment:
Ah, thanks. I was confused as to why we were checking the duration / number cases.

although nit: `s/ValueType/Value/`